tests/Integration/Behaviour/Features/Scenario/Discount/FO/discount_with_code.feature

```diff
@@ -0,0 +1,113 @@
+    And cart "dummy_cart" total with tax included should be '$40.00'
+    And my cart "dummy_cart" should have the following details:
+      | total_products | $40.00 |
+      | shipping       | $7.00  |
+      | total_discount | -$7.00 |
+      | total          | $40.00 |
+
+  Scenario: Create a discount with free shipping without a promo code
+    And there is a product in the catalog named "product1" with a price of 10.00 and 1000 items in stock
+    When I create a "free_shipping" discount "free_shipping_discount_with_minimum_amount_auto" with following properties:
+      | name[en-US] | Minimum amount gives free shipping  |
+      | name[fr-FR] | Montant minimum donne envoi gratuit |
+      | active      | true                                |
+      | valid_from  | 2025-01-01 11:05:00                 |
+      | valid_to    | 2025-12-01 00:00:00                 |
+      | code        |                                     |
+    When I update discount "free_shipping_discount_with_minimum_amount_auto" with the condition of a minimum amount:
+      | minimum_amount                   | 40.00 |
+      | minimum_amount_currency          | usd   |
+      | minimum_amount_tax_included      | true  |
+      | minimum_amount_shipping_included | true  |
+    And discount "free_shipping_discount_with_minimum_amount_auto" should have the following properties:
+      | name[en-US]                      | Minimum amount gives free shipping  |
+      | name[fr-FR]                      | Montant minimum donne envoi gratuit |
+      | active                           | true                                |
+      | valid_from                       | 2025-01-01 11:05:00                 |
+      | valid_to                         | 2025-12-01 00:00:00                 |
+      | code                             |                                     |
+      | minimum_amount                   | 40.00                               |
+      | minimum_amount_currency          | usd                                 |
+      | minimum_amount_tax_included      | true                                |
+      | minimum_amount_shipping_included | true                                |
+    Given I create an empty cart "dummy_cart" for customer "testCustomer"
+    When I add 3 product "product1" to the cart "dummy_cart"
+    And cart "dummy_cart" total with tax included should be '$37.00'
+    And my cart "dummy_cart" should have the following details:
+      | total_products | $30.00 |
+      | shipping       | $7.00  |
+      | total_discount | $0.00  |
+      | total          | $37.00 |
+    When I add 1 product "product1" to the cart "dummy_cart"
+    And cart "dummy_cart" total with tax included should be '$40.00'
+    And my cart "dummy_cart" should have the following details:
+      | total_products | $40.00 |
+      | shipping       | $0.00  |
+      | total_discount | $0.00  |
+      | total          | $40.00 |
```

Comment:
Sometyhing doesn't seem right here, you should have the same details as the above scenario except the discount is automatic and applied with using a code:
```suggestion
    And my cart "dummy_cart" should have the following details:
      | total_products | $40.00 |
      | shipping       | $7.00  |
      | total_discount | -$7.00  |
      | total          | $40.00 |
```